components/workspacekit/cmd/rings.go

```diff
@@ -1072,15 +1074,26 @@ func startInfoService(socketDir string) (func(), error) {
 	}, nil
 }
 
+var lastWorkspaceInfo *api.WorkspaceInfoResponse
+
 func (svc *workspaceInfoService) WorkspaceInfo(ctx context.Context, req *api.WorkspaceInfoRequest) (*api.WorkspaceInfoResponse, error) {
 	client, err := connectToInWorkspaceDaemonService(ctx)
 	if err != nil {
-		return nil, err
+		log.WithError(err).Error("could not connect to workspace daemon")
+		return nil, status.Error(codes.Internal, "could not resolve workspace info")
```

Comment:
```suggestion
		return nil, status.Error(codes.Internal, "could not connect to workspace service")
```
I would prefer to have a different error message here so that we can distinguish between not being able to connect to the service and the service returning an error.

Reply:
This is an error for supervisor to avoid leaking it to users in debug mode. We can have look at workspace kit logs instead to distinguish, ie see logging. It already using different messages.